### ISSUE TITLE
ExcludeVersion switch is not working on CentOS and Ubuntu

### ISSUE THREAD

**Issue by bmanikm** [OneGet/oneget#320]
For more details please take a look at the below reported issues
- PowerShell/PowerShellGet#192
- PowerShell/PowerShell#3986

**Comment by bmanikm** [comment]
Getting below error for `[System.Lazy[System.Boolean]]$true` on WSL - Ubuntu 18.04 OS with 6.1.0-preview.4

### On WSL - Ubuntu 18.04 OS with 6.1.0-preview.4
```powershell
PS /> [System.Lazy[System.Boolean]]$false
Cannot convert value "False" to type "System.Lazy`1[System.Boolean]". Error: "Invalid cast from 'System.Boolean' to 'System.Lazy`1[[System.Boolean, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]'."
At line:1 char:1
+ [System.Lazy[System.Boolean]]$false
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidArgument: (:) [], RuntimeException
+ FullyQualifiedErrorId : InvalidCastIConvertible

PS /> [System.Lazy[System.Boolean]]$true
Cannot convert value "True" to type "System.Lazy`1[System.Boolean]". Error: "Invalid cast from 'System.Boolean' to 'System.Lazy`1[[System.Boolean, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]'."
At line:1 char:1
+ [System.Lazy[System.Boolean]]$true
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidArgument: (:) [], RuntimeException
+ FullyQualifiedErrorId : InvalidCastIConvertible

PS /> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.1.0-preview.4
PSEdition                      Core
GitCommitId                    6.1.0-preview.4
OS                             Linux 4.4.0-17134-Microsoft #137-Microsoft Thu Jun 14 18:46:00 PST 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### On Windows PowerShell 5.1
```powershell
PS C:\> [System.Lazy[System.Boolean]]$false

IsValueCreated Value
-------------- -----
          True False


PS C:\> [System.Lazy[System.Boolean]]$true

IsValueCreated Value
-------------- -----
          True False


PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.17134.165
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17134.165
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by edyoung** [comment]
This is fixed by https://github.com/OneGet/oneget/pull/451

**Comment by edyoung** [comment]
fixed in 2.1.4/packagemanagement 1.4
